### PR TITLE
Feature/02 roles

### PR DESCRIPTION
**Pull request by kdhrubo**

### CODE REVIEW

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/business/RolesBusinessDelegate.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	
+//	@Transactional(transactionManager = "roleTransactionManager")
+	public void register(Role role)
+	{
+		log.info("Role table is updated with role ::", role);
+		roleRepository.save(role);
+		
+	}
+	
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```
Comment (kdhrubo) [inline orderflo/src/main/java/co/hooghly/commerce/orderflo/business/RolesBusinessDelegate.java]: Useless method

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/business/RolesBusinessDelegate.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+import co.hooghly.commerce.orderflo.domain.Role;
+import co.hooghly.commerce.orderflo.roles.repository.RoleRepository;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@Component
+public class RolesBusinessDelegate implements UserDetailsService{
```
Comment (kdhrubo) [inline orderflo/src/main/java/co/hooghly/commerce/orderflo/business/RolesBusinessDelegate.java]: This class need not implement UserDetailsService

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/OrderfloApplication.java [hunk 3]
```diff
@@ -1,8 +1,11 @@
 package co.hooghly.commerce.orderflo;
 
 import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+
+@EnableAutoConfiguration
```
Comment (kdhrubo) [inline orderflo/src/main/java/co/hooghly/commerce/orderflo/OrderfloApplication.java]: This annotation is not required from Spring Boot 1.2 onwards. @SpringApplication wraps 3 annotations including @EnableAutoConfiguration.

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.stereotype.Component;
+
+import co.hooghly.commerce.orderflo.domain.User;
+import co.hooghly.commerce.orderflo.roles.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@Component
+public class UserBusinessDelegate implements UserDetailsService{
```
Comment (kdhrubo) [inline orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java]: This should extend the abstract base class and also implement the user details service to cater to Spring security requirements.

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+	@Autowired
+	private UserRepository userRepository;
+	
+//	@Transactional(transactionManager = "roleTransactionManager")
+	public void register(User user)
```
Comment (kdhrubo) [inline orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java]: The save method of the abstract base class should be called via inheritance.

File: orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java (hunk 4 above)
Comment (Rajiv-mar121) [reply orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java]: This interfcae "UserDetailsService " is required ?
In earlier comment you told that it is not required ? if needed then below method has to override .

Comment (kdhrubo) [reply orderflo/src/main/java/co/hooghly/commerce/orderflo/business/UserBusinessDelegate.java]: @Rajiv-mar121 That is applicable for RoleBusinessDelegate. UserBusinessDelegate it is required.

**Comment by Rajiv-mar121** [comment]
Hi Dhrubo,

Please review once . Code refactored

feature/02_roles

//Rajiv


On Tue, Dec 26, 2017 at 3:12 PM, Dhrubojyoti Kayal <notifications@github.com

**Pull request merged**